api/app/Http/Controllers/EntityController.php

```diff
@@ -116,30 +131,31 @@ public function putMany(Request $request)
     {
         $requestCollection = $request->data;
 
-        $ids = $this->getIds($requestCollection, $this->getKeyName(), $this->validateRequestRule);
+        $ids = $this->getIds($requestCollection, $this->getModel()->getKeyName(), $this->validateRequestRule);
         $models = [];
         if (!empty($ids)) {
-            $models = $this->getRepository()->findMany($ids);
+            $models = $this->getManyEntitiesByIds($ids);
         }
 
         $putModels = [];
+        $keyName = $this->getModel()->getKeyName();
         foreach ($requestCollection as $requestEntity) {
-            $id = isset($requestEntity[$this->getKeyName()])?$requestEntity[$this->getKeyName()]:null;
+            $id = isset($requestEntity[$keyName])?$requestEntity[$keyName]:null;
```

Comment:
This `foreach` loop should be extracted out to a method so that inheriting controllers can override this method but still call `$this->findOrNewCollection($requestCollection)`